src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -129,7 +129,7 @@ public ResettableSqlStatement GetResettableStatement(string query)
             return new ResettableSqlStatement(statement);
         }
 
-        public void RunInTransaction(Action action)
+        public TResult RunInTransaction<T, TResult>(T userData, Func<T, TResult> action)
```

Comment:
📝 The equivalent of this change was merged in #36127, largely due to the groundwork laid by this pull request